src/main/java/io/confluent/idesidecar/restapi/connections/ConnectionState.java

```diff
@@ -183,4 +188,19 @@ public Optional<Credentials> getKafkaCredentials() {
   public Optional<Credentials> getSchemaRegistryCredentials() {
     return Optional.empty();
   }
+
+  /**
+   * Get the TLS configuration to use when connecting to the cluster (Kafka or Schema Registry). We
+   * assume that the TLS configuration is the same for both Kafka and Schema Registry. In the
```

Comment:
Actually, it's not clear whether or not SR supports mTLS for clients. That is an old issue that could have been not updated correctly. We need to verify with the Governance team, because the documentation does allude to it working.

[This section of the docs](https://docs.confluent.io/platform/current/security/authorization/rbac/configure-mtls-rbac.html#configure-sr) talks about how to enable mTLS in the Schema Registry config for CP. That's the server side config, and [cp-demo is definitely not configured to enable mTLS](https://github.com/confluentinc/cp-demo/blob/f0422a964fe8d111601df0d795fe774299da4b08/docker-compose.yml#L799-L804) since `ssl.client.authentication=NONE`, and it is slightly different than [this section that talks about configuring SR to use mTLS against Kafka](https://docs.confluent.io/platform/current/security/authentication/mutual-tls/overview.html#sr) (note the `kafkastore` prefix, which covers the configuration of Kafka clients that SR uses to talk with Kafka).

The [Schema Registry client configuration reference](https://docs.confluent.io/platform/current/schema-registry/sr-client-configs.html#ssl-keystore-location) shows this for `ssl.keystore.location` (emphasis mine):
> The location of the key store file. This is optional for client and can be used for *two-way authentication for client*.

The [Confluent Platform 7.8 announcement](https://www.confluent.io/blog/introducing-confluent-platform-7-8/#scale-platform-security-by-enabling-rbac-authorization-with-mtls-identity) says this (emphasis mine):
> With this enhancement, identities extracted from mTLS authentication can now be used to enforce RBAC authorization directly within the platform. This advancement allows customers to adopt RBAC without compromising their existing mTLS security framework, enabling both service-to-service authentication and granular access control. This is now **generally available across all Confluent Platform services** and includes supporting mTLS authentication to the Confluent Metadata Service.

[MDS clients](https://docs.confluent.io/platform/current/security/authorization/rbac/configure-mtls-rbac.html#configure-mds-client) can now also use mTLS.

The [release notes](https://docs.confluent.io/platform/current/release-notes/index.html#security) also says this under the "Security" section, but also this under the ["Schema Registry" section](https://docs.confluent.io/platform/current/release-notes/index.html#sr):
> RBAC can be enabled without requiring an identity provider by using mTLS identity for RBAC authorization for Schema Registry in Confluent Platform 7.8. For more information, see [Use mTLS Authentication with RBAC Authorization in Confluent Platform](https://docs.confluent.io/platform/current/security/authorization/rbac/mtls-rbac.html#mtls-rbac).

And finally, [this page](https://docs.confluent.io/platform/current/security/authorization/rbac/mtls-rbac.html#mtls-rbac) says:
> All the Confluent Platform REST interfaces accept TLS certificates for authentication from which the principal for RBAC authorization can be extracted from.